db.js

```diff
@@ -449,21 +449,56 @@ var loans = (function () {
       let updateInserts = [loan.status, loan.id];
       updateQuery = mysql.format(updateQuery, updateInserts);
 
-      //var now = moment().format("YYYY-MM-DD HH:mm:ss");
-      //will be used later for timestamps
-      //console.log(now);
-
-      return pool.query(updateQuery).then((updateResult) => {
-        if (loan.status === "accepted") {
-          let deleteQuery =
-            "DELETE FROM loans WHERE status = 'pending' AND book_id = ?";
-          let deleteInserts = [loan.book_id];
-          deleteQuery = mysql.format(deleteQuery, deleteInserts);
-
-          pool.query(deleteQuery);
-        }
-        return updateResult;
-      });
+      var currentDate = moment().format("YYYY-MM-DD HH:mm:ss");
+
+      var updateTimeQuery = "UPDATE loans SET ";
+      // After createing an update query, fetching the current time, and creating a starting updateTime query,
+      // check to see if the status is anything but "accepted." If so, set the time query to update its respective field.
+
+      if (loan.status === "loaned") {
+        updateTimeQuery = updateTimeQuery + "loan_start_date = ? WHERE id = ?";
+        let updateTimeInserts = [currentDate, loan.id];
+        updateTimeQuery = mysql.format(updateTimeQuery, updateTimeInserts);
+
+        pool.query(updateTimeQuery);
+      } else if (loan.status === "returned") {
+        updateTimeQuery = updateTimeQuery + "return_date = ? WHERE id = ?";
+        let updateTimeInserts = [currentDate, loan.id];
+        updateTimeQuery = mysql.format(updateTimeQuery, updateTimeInserts);
+
+        pool.query(updateTimeQuery);
+      } else if (loan.status === "canceled" || loan.status === "declined") {
```

Comment:
So maybe I misspoke here. When I said that the "declined" and "canceled" delete the loan I meant that on the frontend the buttons labeled "Delete" and "Cancel" actually call the DELETE route not the PUT route. So they don't need any special side effects here. The only side effects should be
- For "accept" set accept date and delete other pending requests
- For "loaned" set loaned date and possibly due date (end date?)

If we were to implement the "canceled" or "declined" state, they would not have special side effects that alter the database.

Reply:
You're all good, no worries! I'll remove the cancel/delete logic. In addition to the 'accept' and 'loaned' state, 'returned' also should remain where it is, right?